### ISSUE TITLE
Is there any deep reason to distribhte the prebuilt binaries without JNI?

### ISSUE THREAD

**Issue by KOLANICH** [oracle/graal#931]

**Comment by christianwimmer** [comment]
No. Right now, we still don't need any JNI methods because all JDK native code is re-implemented in Java. That will change soon, and then JNI will be a necessary part of every image.

What exactly do you want to do via JNI? Unless extra information is registered at image build time, the only thing you can do via JNI is to create and enter a VM using the JNI attach API functions.

**Comment by KOLANICH** [comment]
>What exactly do you want to do via JNI?

I want to use GraalVM instead of OpenJDK.

More precisely, graalpython doesn't look like ready for any use except of testing. But testing on it may be still desireable.

To bridge CPython to Java (import and use classes of JVM langs from cpython) there exist a package JPype using JNI, and I wonder if I can use it to bridge CPython to GraalVM, so I won't have to have 2 impls of JVM (OpenJDK and Graal) in my CI docker images.

Is it possible? Is Graals' NDK a drop-in replacement for OpenJDK NDK?

**Comment by christianwimmer** [comment]
@timfel can answer Python questions better than me.

I don't know exactly what you mean with "NDK", that seems to be an Android-specific thing. In general, GraalVM contains a full JDK. So you can definitely use GraalVM and GraalPython in `--jvm` mode to do anything using JNI. To make a Python-Java bridge via JNI working on a native image, I would assume that quite a good number of Java functions need to be registered for JNI usage at image build time.
